src/main/java/org/avaje/agentloader/AgentLoader.java

```diff
@@ -151,6 +175,68 @@ private static final boolean isWindows() {
     return File.separatorChar == '\\';
   }
 
+  /**
+   * This method takes the jar:file:path-to-filename.war!/WEB-INF/jar/jar-file/ offset that is included in the
+   * url classpath and extracts out a single jar containing the files in that match that url.
+   *
+   * @param path    - full url entry in the classpath
+   * @param agentName - the agent name that we are trying to match
+   * @return null if it fails or a full path to the jar file if it succeeds
+   */
+  public static String extractJar(URL path, String agentName) {
+    String fullPath = null;
+
+    String[] jarNames = path.getPath().split(":");
+
+    if (jarNames.length >= 2) {
```

Comment:
*> BTW, do you think it is OK for me to switch the logger to slf4j instead
of logging?*

So yes that would add a dependency but on the other hand it is the
AgentLoader (not the agent) so really AgentLoader would typically be
considered part of the application (and not standalone like the agent) so
yes I think we could change that over to SLF4J.

... but let me ponder that a little bit more.

Note that if we change the dependency I think we should bump the major
version number.


On 2 July 2015 at 11:05, Kefeng DENG <notifications@github.com> wrote:

> In src/main/java/org/avaje/agentloader/AgentLoader.java
> <https://github.com/avaje-common/avaje-agentloader/pull/3#discussion_r33734313>
> :
>
> > @@ -151,6 +175,68 @@ private static final boolean isWindows() {
> >      return File.separatorChar == '\\';
> >    }
> >
> > +  /**
> > +   * This method takes the jar:file:path-to-filename.war!/WEB-INF/jar/jar-file/ offset that is included in the
> > +   * url classpath and extracts out a single jar containing the files in that match that url.
> > +   *
> > +   * @param path    - full url entry in the classpath
> > +   * @param agentName - the agent name that we are trying to match
> > +   * @return null if it fails or a full path to the jar file if it succeeds
> > +   */
> > +  public static String extractJar(URL path, String agentName) {
> > +    String fullPath = null;
> > +
> > +    String[] jarNames = path.getPath().split(":");
> > +
> > +    if (jarNames.length >= 2) {
>
> Thanks for your comments. BTW, do you think it is OK for me to switch the
> logger to slf4j instead of logging?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/avaje-common/avaje-agentloader/pull/3/files#r33734313>
> .
>